packages/happy-dom/src/event/EventTarget.ts

```diff
@@ -32,7 +41,8 @@ export default abstract class EventTarget implements IEventTarget {
 		listener: ((event: Event) => void) | IEventListener
 	): void {
 		if (this._listeners[type]) {
-			const index = this._listeners[type].indexOf(listener);
+			const _listeners = this._listeners[type].map((l) => l.fn);
```

Comment:
For loops are faster than map(), so I think we are better of with a for loop here, but we could improve performance even further if we store options and listeners in separate objects.

Example:
```typescript
public readonly _options: {
	[k: string]: { once: boolean }[];
} = {};
public readonly _listeners: {
	[k: string]: (((event: Event) => void) | IEventListener)[];
} = {};

public addEventListener(
	type: string,
	listener: ((event: Event) => void) | IEventListener,
	options?: { once: boolean }
): void {
	this._listeners[type] = this._listeners[type] || [];
	this._options[type] = this._options[type] || [];
	this._listeners[type].push(listener);
	this._options[type].push(options || null);
}
```

Reply:
Great.